server/src/main/java/org/opensearch/index/compositeindex/datacube/ReadDimension.java

```diff
@@ -26,15 +26,24 @@ public class ReadDimension implements Dimension {
     public static final String READ = "read";
     private final String field;
     private final DocValuesType docValuesType;
+    private final ComparisonType comparisonType;
 
     public ReadDimension(String field) {
         this.field = field;
         this.docValuesType = DocValuesType.SORTED_NUMERIC;
+        this.comparisonType = ComparisonType.SIGNED;
     }
 
     public ReadDimension(String field, DocValuesType docValuesType) {
         this.field = field;
         this.docValuesType = docValuesType;
+        this.comparisonType = ComparisonType.SIGNED;
+    }
+
+    public ReadDimension(String field, ComparisonType comparisonType) {
```

Comment:
We will need to add this rather : 
```
    public ReadDimension(String field, DocValuesType docValuesType, ComparisonType comparisonType) {
```

Dimensions always get initialized with field and DvType

Reply:
Done.